AEPCampaignClassic/Sources/RegistrationManager.swift

```diff
@@ -111,29 +111,31 @@ class RegistrationManager {
                              systemInfoService.getFormattedLocale().urlEncoded)
         payload.append(additionalParametersXML)
 
-        /// build url
+        // build url
         let urlString = String(format: CampaignClassicConstants.REGISTRATION_API_URL_BASE, marketingServer)
         guard let url = URL(string: urlString) else {
             Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration failed, Invalid network request URL : \(urlString)")
             return
         }
 
-        /// build headers
+        // build headers
         let headers = buildHeaders(payload: payload)
 
-        /// create a network request
+        // create a network request
         let request = NetworkRequest(url: url, httpMethod: .post, connectPayload: payload, httpHeaders: headers, connectTimeout: configuration.timeout, readTimeout: configuration.timeout)
 
         Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration request initiated with \n URL : \(url.absoluteString) \n Headers: \(headers) \n Payload : \(payload)")
 
-        /// make the network request
+        // make the network request
         ServiceProvider.shared.networkService.connectAsync(networkRequest: request) { connection in
             guard connection.responseCode == 200 else {
                 Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration failed, Network Error. Response Code: \(String(describing: connection.responseCode)) URL : \(url.absoluteString)")
                 return
             }
 
-            Log.debug(label: CampaignClassicConstants.LOG_TAG, "Device Registration success. Saving the hashed registration data.")
+            // retrieve the response message from the body
+            let responseMessage = String(data: connection.data ?? Data(), encoding: .utf8) ?? ""
```

Comment:
here (and above), maybe it's better to have some canned string if we can't get the response message from the connection's data?  something like "Response message : unable to get response message"?